### DISCUSSION TITLE
Hardware Linux

### DISCUSSION THREAD

**Discussion by sp00ck** [markqvist/Reticulum#228] [General]
The adopted development model forces the use of a full-fledged operating system or moving everything to a platform large enough to run normal python3 there. 
That's why I think it's necessary to create a Linux computer based on wifi mesh.
Despite many technologies like batman, mesh networks don't work because they are just difficult. Reticulum promises an easier setup. 

If someone can show a mesh network configuration that doesn't require dns addressing, doesn't require ip address assignment and will work I think it will be something that will popularize the project.

**Comment by markqvist** [comment]
> Despite many technologies like batman, mesh networks don't work because they are just difficult. Reticulum promises an easier setup.

Yes, this is definitely the goal here, while also solving the problems of scalability, security and privacy.

> The adopted development model forces the use of a full-fledged operating system

Currently, you need *some sort of environment* that can run Python 3. That's a very low requirement. You don't need a "full operating system" to do this. You can spin up RNS on a bare-metal kernel and a few extra binaries. It works perfectly and is extremely light.

For running a full RNS stack on *really* limited devices, like low-power microcontrollers, as I find myself repeating, please have a look at [the Roadmap](https://github.com/markqvist/Reticulum/blob/master/Roadmap.md). It's pretty clear on how these things are going to be acheived.

> If someone can show a mesh network configuration that doesn't require dns addressing, doesn't require ip address assignment and will work I think it will be something that will popularize the project.

This is already possible, right now. It's all in the manual and examples if you read it. You can set it up in 5 minutes.

**Reply by kc1awv** [reply]
> For running a full RNS stack on really limited devices, like low-power microcontrollers

In fact, there's a stack of Pine64 Ox64 on my bench with a few reserved for this...

**Reply by markqvist** [reply]
Pretty cool. Looking forward to hearing about what you can achieve with them!

**Comment by fab1an2** [comment]
>>If someone can show a mesh network configuration that doesn't require dns addressing, doesn't require ip address assignment 
>>and will work I think it will be something that will popularize the project.

>This is already possible, right now. It's all in the manual and examples if you read it. You can set it up in 5 minutes.

Python alone is not enough, you must also have a network and some sort of file system.

Please show which guide allows you to use wifi as a transport layer. I would love to do something like this. ESP 6288 will do?

**Reply by Swissbandit** [reply]
if the device rns and sb/nomad ist running also has wifi capabilities,
activate the Autointerface and every sideband or nomad device capable of connecting to this wifi will be able to use the system if autointerface is active there aswell.

the limiting factor atm is that a device has to support rns in some way,or needs a helping part like a raspi attached,until the microphyton implementation is published,witch will make the hardware even more lightweight to produce repeaters and gateways.

**Reply by Swissbandit** [reply]
there are some new versions of wifi devices having enough space to put rns on libremesh into action but this is at experimental stage.
of course if more people work in this direction,more will become known and documented over time.
